dashboard/src/main/resources/templates/component.ftl

```diff
@@ -24,21 +27,42 @@
     
       <ul>
         <#list globalUpperBoundFailures as lower, upper>
-          <li>Upgrade ${lower} to ${upper}:
-          
-          <p>Add this dependency element to the pom.xml for ${groupId}:${artifactId}:${version}:</p>
-          
-<pre class="suggested-dependency-mediation"><code>&lt;dependency>
-  &lt;groupId>${upper.getGroupId()}&lt;/groupId>
-  &lt;artifactId>${upper.getArtifactId()}&lt;/artifactId>
-  &lt;version>${upper.getVersion()}&lt;/version>
+          <#if lower == artifact>
```

Comment:
What's the equals semantics here? identity or equality?

Reply:
It's equality ([Freemarker Expressions - Comparison](https://freemarker.apache.org/docs/dgui_template_exp.html#dgui_template_exp_comparison)). But I updated the code to compare only the coordinates, without 'file' attribute of Artifact instance.